tests/test_flows.py

```diff
@@ -1033,27 +1033,26 @@ async def test_subflow_timeout_waits_until_execution_starts(self, tmp_path):
         canary_file = tmp_path / "canary"
 
         @flow(timeout_seconds=1)
-        def downstream_flow():
+        async def downstream_flow():
             canary_file.touch()
 
         @task
-        def sleep_task(n):
-            time.sleep(n)
+        async def sleep_task(n):
+            await anyio.sleep(n)
 
         @flow
         async def my_flow():
-            upstream_sleepers = sleep_task.map(list(range(3)))
-            downstream_flow(wait_for=upstream_sleepers)
+            upstream_sleepers = await sleep_task.map([0.5, 1.0])
+            await downstream_flow(wait_for=upstream_sleepers)
 
         t0 = anyio.current_time()
         state = await my_flow._run()
         t1 = anyio.current_time()
 
         assert state.is_completed()
 
-        # Validate the sleep tasks have ran.
-        # Note: t1 - t0 can be less than exactly 3 (i.e., around 2.9). By comparing with 2.7 we have some leeway.
-        assert t1 - t0 >= 2.7
+        # Validate the sleep tasks have ran
+        assert t1 - t0 >= 1
```

Comment:
Noted in the commit message, but this test was wrong before. The `range(3)` did not include a `sleep(3)` so the target time was 2s not 3s.